homeassistant/components/sia/sia_entity_base.py

```diff
@@ -2,51 +2,54 @@
 from __future__ import annotations
 
 from abc import abstractmethod
-import logging
-from typing import Any
 
 from pysiaalarm import SIAEvent
 
-from homeassistant.config_entries import ConfigEntry
-from homeassistant.const import CONF_PORT
 from homeassistant.core import CALLBACK_TYPE, State, callback
 from homeassistant.helpers.dispatcher import async_dispatcher_connect
 from homeassistant.helpers.entity import DeviceInfo
 from homeassistant.helpers.event import async_call_later
 from homeassistant.helpers.restore_state import RestoreEntity
 
-from .const import CONF_ACCOUNT, CONF_PING_INTERVAL, DOMAIN, SIA_EVENT, SIA_NAME_FORMAT
-from .utils import get_attr_from_sia_event, get_unavailability_interval
-
-_LOGGER = logging.getLogger(__name__)
+from .const import DOMAIN, LOGGER, SIA_EVENT, SIA_HUB_ZONE
+from .utils import (
+    SIAAlarmControlPanelEntityDescription,
+    SIABinarySensorEntityDescription,
+    get_attr_from_sia_event,
+    get_name,
+    get_unavailability_interval,
+)
 
 
 class SIABaseEntity(RestoreEntity):
     """Base class for SIA entities."""
 
+    entity_description: SIAAlarmControlPanelEntityDescription | SIABinarySensorEntityDescription
```

Comment:
```suggestion
    entity_description: SIARequiredKeysMixin
```

Reply:
it also relies on the fields in the two other entity descriptions (like device class) so not sure if this works...

Reply:
Tried it, and that doesn't work for Mypy because device_class is not part of SIARequiredKeysMixin (it comes in through the alarm_control_panel and binary_sensor ED's

Reply:
Try to create an intermediate class here, that the others inherit inside the platform.
```suggestion
    entity_description: SIABaseEntityDescription
```

Reply:
Or in utils

Reply:
Hard to review on mobile. Will take a look over the WE unless someone beats me to it.

Reply:
I really think that it would be better to move `SIARequiredKeysMixin` here, and create a new `SIAEntityDescription` which just inherits `SIARequiredKeysMixin`.
Then finally `SIAAlarmControlPanelEntityDescription` and `SIABinarySensorEntityDescription` inherit `SIAEntityDescription` instead of the `SIARequiredKeysMixin`

Reply:
yeah, got it now, should be easier to read and understand!